### PR TITLE
Mixer documentation 

### LINKED ISSUE #100: Document Mixer
- Document the Mixer module to Doxy compatible javadoc standard.

Filter.hh and the [doxy manual](https://www.doxygen.nl/manual/docblocks.html) may be useful.

### PR DESCRIPTION
**Pull request by bear-in-the-air**
closes #100

### CODE REVIEW

File: include/mixer/mixer.hh [hunk 1]
```diff
@@ -11,33 +11,59 @@
...
+ */
 template <std::size_t num_banks> class Mixer {
   private:
+    // Array of pipes for the input to the class
     std::array<Pipe<Audio>, num_banks> &input_pipes;
+    // Single pipe as output for the class
     Pipe<Audio> &output_pipe;
 
+    // Not sure what this does, can't find any other reference to it
     bool run_thread{};
```
Comment (bear-in-the-air) [inline include/mixer/mixer.hh:31]: I don't know what this does, I don't see any other reference to it in the file.

Comment (Jambls) [reply include/mixer/mixer.hh:31]: Probably left over from before we switched to using wait_for()

**Review by Jambls — APPROVED** [review]
Looks good to me

File: include/rms/rms.hh [hunk 2]
```diff
@@ -46,22 +46,22 @@ template <std::size_t num_samples> class RMS {
...
 
   public:
-    /** 
-    *   Initialises the input and output pipes, and the timeout.
-    *   @param input_pipe A Pipe<Audio> containing the signal to be measured
-    *   @param output_pipe A Pipe<double> which will contain the root mean square of the signal
-    *   @param timeout The number of milliseconds the thread should wait before exiting if no data is provided. 
-    */
+    /**
+     *   Initialises the input and output pipes, and the timeout.
```
[the remark this answers is not in the document]
Comment (MHCooke) [reply include/rms/rms.hh:50]: but it passed ci... more likely angus' IDE is configured to remove extra spaces?

[the remark this answers is not in the document]
Comment (MHCooke) [reply include/rms/rms.hh:50]: either way, if rms.hh does need fixed it should be done in a separate pr imo

File: include/mixer/mixer.hh (hunk 1 above)
Comment (MHCooke) [reply include/mixer/mixer.hh:31]: Good idea to remove it then imo. Why doesn't SA or the compiler complain about an unused variable though.

File: include/mixer/mixer.hh [hunk 3]
```diff
@@ -11,33 +11,59 @@
...
+     */
     Mixer(std::array<Pipe<Audio>, num_banks> &input_pipes_, Pipe<Audio> &output_pipe_,
           std::chrono::milliseconds timeout_)
         : input_pipes(input_pipes_), output_pipe(output_pipe_), timeout(timeout_) {
     }
 
     // explicitly disable copy and move constructors since that will mess
     // with threading logic. if these are needed it can be reviewed later
+
+    // explicitly delete the copy constructor
```
Comment (MHCooke) [inline include/mixer/mixer.hh]: These are not javadoc comments and don't appear in the doxygen

File: include/mixer/mixer.hh [hunk 4]
```diff
@@ -11,33 +11,59 @@
...
     Mixer(const Mixer &) = delete;
+    // explicitly delete copy assignment constructor
     Mixer &operator=(const Mixer &) = delete;
+    // explicitly delete move constructor
     Mixer(Mixer &&) = delete;
+    // explicitly delete move assignment constructor
     Mixer &operator=(Mixer &&) = delete;
 
+    /**
+     * Sums the samples at maching points on each sub array and combines it in to a single array of samples
```
Comment (MHCooke) [inline include/mixer/mixer.hh:70]:
```suggestion
     * Sums the samples at matching points on each sub array and combines it in to a single array of samples
```

File: include/mixer/mixer.hh [hunk 5]
```diff
@@ -78,7 +106,7 @@ template <std::size_t num_banks> class Mixer {
             return true;
         });
     }
-
+    // Join the mixer thread when the destructor is called
```
Comment (MHCooke) [inline include/mixer/mixer.hh]: also not javadoc

File: include/rms/rms.hh (hunk 2 above)
[the remark this answers is not in the document]
Comment (bear-in-the-air) [reply include/rms/rms.hh:50]: I don't think I have anything set up that modifies white space aside from adding a new line on the end of the file on save.

I rolled back the commit to previous, then ran checks.sh and it comes back to this version, so I think this is the post checks.sh version and it wasn't run on the merged code.

Not sure however why the CI didn't catch it. Perhaps the auto format doesn't match the error only checks.

**Review by MHCooke — APPROVED** [review]

**Pull request merged**